skimage/restoration/deconvolution.py

```diff
@@ -376,17 +420,65 @@ def richardson_lucy(image, psf, iterations=50, clip=True):
     else:
         convolve_method = convolve
 
+    if return_iterations:
+        all_iterations = np.empty((iterations, 2,) + image.shape)
+
     image = image.astype(np.float)
-    psf = psf.astype(np.float)
+
+    if psf is None:
+        psf = np.full(image.shape, 0.5)
+    elif psf is not None and blind:
+        assert psf.shape == image.shape, \
+            'For blind deconvolution, ' \
+            'image and PSF should have the same shape!'
+        psf = psf.astype(np.float)
+    else:
+        pass
```

Comment:
I'm not sure to understand the logic...
From the docstring, I would have (inconsistently) understood that: 
psf = None -> blind deconvolution
psf != None -> already implemented behavior.
Why do we need a `blind` boolean?

I think it needs more clarifications...

Reply:
@sciunto thanks for checking the code, I'll work on the comments.
Regarding the API: 
If psf = None -> blind deconv, this is true.
However, you could also pass a psf as an initial estimate, that should be refined each iteration.
Maybe rename the `blind` keywoord to `refine_psf`?

Reply:
OK, got it, now I see the threefold input. Let's think about it. I'll continue to read the code with this in mind.